storage/src/integration-test/java/tech/pegasys/teku/storage/server/kvstore/DatabaseTest.java

```diff
@@ -2435,21 +2435,21 @@ public void pruneAllSidecars_isOperative(final DatabaseContext context) throws I
     database.addSidecar(dataColumnSidecar1_1);
     database.addSidecar(dataColumnSidecar2_0);
 
-    database.pruneAllSidecars(ZERO);
+    database.pruneAllSidecars(ZERO, 10);
     try (final Stream<DataColumnSlotAndIdentifier> dataColumnIdentifiersStream =
         database.streamDataColumnIdentifiers(ZERO, dataColumnSidecar2_0.getSlot())) {
       assertThat(dataColumnIdentifiersStream.toList())
           .containsExactly(
               columnSlotAndIdentifier1_0, columnSlotAndIdentifier1_1, columnSlotAndIdentifier2_0);
     }
 
-    database.pruneAllSidecars(ONE);
+    database.pruneAllSidecars(ONE, 10);
     try (final Stream<DataColumnSlotAndIdentifier> dataColumnIdentifiersStream =
         database.streamDataColumnIdentifiers(ZERO, dataColumnSidecar2_0.getSlot())) {
       assertThat(dataColumnIdentifiersStream.toList()).containsExactly(columnSlotAndIdentifier2_0);
     }
 
-    database.pruneAllSidecars(dataColumnSidecar2_0.getSlot());
+    database.pruneAllSidecars(dataColumnSidecar2_0.getSlot(), 10);
```

Comment:
Could we also have separate test with, say, 3 slot sidecars (more than 1 sidecar at lease on one slot) and pruneLimit 2, checking that pruneLimit actually works, please.